### PR TITLE
Drop obsolete platforms

### LINKED ISSUE #4347: Dropping obsolete platforms
We currently have an important number of platforms in ``platform/``, but some of those have hardly been supported over the last two year, probably don't work and refer to somewhat obsolete technologies. Those would be:

- flash: @punto- hinted on IRC that this probably never really worked, it contains [nonfree files](https://github.com/godotengine/godot/blob/master/platform/flash/Console.as), and well, Flash should die and I'd be for not even suggesting that we might be working on it. So: **drop**.
- nacl: I don't know if anyone even tried to build it since the opensourcing. NACL itself is still being somewhat developed, but should we spend time on it or just ditch it [Waiting for WebAssembly™](https://hacks.mozilla.org/2016/03/a-webassembly-milestone/)? I'd propose to **drop**, but if others would like to keep it, why not.

Then we have:
- bb10: The personal favourite of @punto- AFAIK ;) I never tried to build it myself, but it seems BB10 is not 100% dead yet, so I guess we can keep it around until @punto- ditches his BB10 phone :p
- winrt: It might be somewhat broken nowadays because it hasn't been much worked on recently, but it should not be dropped, we need to add proper support for UWP.

WDYT? If there's no objection, I'd drop ``platform/flash`` and ``platform/nacl``.

### PR DESCRIPTION
**Pull request by akien-mga**
As discussed in #4347, dropping the Flash and NACL platforms, that are no longer maintained.

Closes #4347.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,10 +1,9 @@
...
+# Godot auto generated files
 platform/server/logo.h
 platform/android/logo.h
 platform/bb10/logo.h
 platform/iphone/logo.h
 platform/javascript/logo.h
-platform/nacl/logo.h
 platform/osx/logo.h
 platform/windows/logo.h
 platform/x11/logo.h
```
Comment (bojidar-bg) [inline .gitignore]: BTW, shouldn't this be `platform/*/logo.h`?

Comment (akien-mga) [inline .gitignore]: It wouldn't hurt yeah. Most important was to fix `s/GoDot/Godot/`. I'm still shivering from it :P

**Comment by akien-mga** [comment]
Note for people compiling the master branch regularly: you'll have to manually delete the build artifacts left over in ``platform/nacl`` and ``platform/flash`` to be able to build Godot from your local clone.

**Comment by mownier** [comment]
You can do:
```
$ git clean -fd
```

**Pull request merged**